server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -711,6 +889,94 @@ boolean blackhole() {
                 assert blackholed || unDisconnected == false;
                 return blackholed;
             }
+
+            void heal() {
+                boolean unDisconnected = disconnectedNodes.remove(localNode.getId());
+                boolean unBlackholed = blackholedNodes.remove(localNode.getId());
+                assert unDisconnected || unBlackholed;
+            }
+
+            ClusterState getLastAppliedClusterState() {
+                return clusterApplier.lastAppliedClusterState;
+            }
+
+            private class FakeClusterApplier implements ClusterApplier {
+
+                final ClusterName clusterName;
+                private final ClusterSettings clusterSettings;
+                ClusterState lastAppliedClusterState;
+
+                private FakeClusterApplier(Settings settings, ClusterSettings clusterSettings) {
+                    clusterName = ClusterName.CLUSTER_NAME_SETTING.get(settings);
+                    this.clusterSettings = clusterSettings;
+                }
+
+                @Override
+                public void setInitialState(ClusterState initialState) {
+                    assert lastAppliedClusterState == null;
+                    assert initialState != null;
+                    lastAppliedClusterState = initialState;
+                }
+
+                @Override
+                public void onNewClusterState(String source, Supplier<ClusterState> clusterStateSupplier, ClusterApplyListener listener) {
+                    switch (clusterStateApplyResponse) {
+                        case SUCCEED:
+                            deterministicTaskQueue.scheduleNow(onNode(localNode, new Runnable() {
```

Comment:
This adds an extra `2 * DEFAULT_DELAY_VARIABILITY` to the maximum time it might take to complete a publication - one delay on a follower and another on the leader, so we need to update `DEFAULT_CLUSTER_STATE_UPDATE_DELAY` accordingly.

Reply:
great catch! fixed in e2a4f41

Reply:
Wasn't me, it was the tests :)